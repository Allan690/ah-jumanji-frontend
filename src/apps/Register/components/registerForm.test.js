import React from 'react';
import Enzyme, {mount, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { MemoryRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import {Provider} from "react-redux";

import RegistrationForm from './registerForm';
import RegisterContainer from "./registerContainer";

Enzyme.configure({ adapter: new Adapter() });

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

describe('<RegisterContainer />', () => {
  it('should match the snapshot', () => {
    const component = shallow(<RegisterContainer debug />);
    expect(component).toMatchSnapshot();
  });
});

describe('Registration form', () => {
  const store = mockStore({});
  const wrapper = mount(
    <Provider store={store}>
      <MemoryRouter>
        <RegistrationForm />
      </MemoryRouter>
    </Provider>,
  );

  it('should match the snapshoot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render register page correctly', () => {
    expect(wrapper.find('.panel-heading').exists()).toBe(true);
    expect(wrapper.find('h3').text()).toBe('Create Account');
    expect(wrapper.find('h3').length).toEqual(1);
    expect(wrapper.find('.panel-body').exists()).toBe(true);
    expect(wrapper.find('TextInput')).toBeDefined();
  });
});
