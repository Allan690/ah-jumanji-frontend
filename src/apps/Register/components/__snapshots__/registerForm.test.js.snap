// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RegisterContainer /> should match the snapshot 1`] = `
<div
  className="row login-container"
>
  <WelcomeSection />
  <Connect(RegistrationForm) />
</div>
`;

exports[`Registration form should match the snapshoot 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <MemoryRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "canGo": [Function],
          "createHref": [Function],
          "entries": Array [
            Object {
              "hash": "",
              "key": "gpkks7",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
          ],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "index": 0,
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "key": "gpkks7",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Connect(RegistrationForm)>
        <RegistrationForm
          actions={
            Object {
              "ActivateUser": [Function],
              "RegisterFail": [Function],
              "RegisterSuccess": [Function],
              "RegisterUser": [Function],
            }
          }
        >
          <div
            className="col-md-6"
          >
            <ToastContainer
              autoClose={5000}
              bodyClassName={null}
              className={null}
              closeButton={
                <CloseButton
                  ariaLabel="close"
                />
              }
              closeOnClick={true}
              draggable={true}
              draggablePercent={80}
              hideProgressBar={false}
              newestOnTop={false}
              pauseOnFocusLoss={true}
              pauseOnHover={true}
              position="top-right"
              progressClassName={null}
              progressStyle={null}
              rtl={false}
              style={null}
              toastClassName={null}
              transition={[Function]}
            >
              <div
                className="Toastify"
              />
            </ToastContainer>
            <div
              className="panel panel-default"
              id="login-form"
            >
              <div
                className="panel-heading"
              >
                <h3>
                  Create Account
                </h3>
                <hr />
              </div>
              <div
                className="panel-body"
              >
                <form>
                  <FormInput
                    inputName="username"
                    inputType="username"
                    label="Username"
                    onChange={[Function]}
                    placeholder="Enter username"
                  >
                    <div
                      className="form-group"
                    >
                      Username
                      <input
                        className="form-control"
                        name="username"
                        onChange={[Function]}
                        placeholder="Enter username"
                        required={true}
                        type="username"
                      />
                      <ValidationSubscript>
                        <p
                          className="subscript subscript-warning hidden"
                        >
                          Kuna kitu umekosea hapa
                        </p>
                      </ValidationSubscript>
                    </div>
                  </FormInput>
                  <FormInput
                    inputName="email"
                    inputType="email"
                    label="Email"
                    onChange={[Function]}
                    placeholder="Enter email"
                  >
                    <div
                      className="form-group"
                    >
                      Email
                      <input
                        className="form-control"
                        name="email"
                        onChange={[Function]}
                        placeholder="Enter email"
                        required={true}
                        type="email"
                      />
                      <ValidationSubscript>
                        <p
                          className="subscript subscript-warning hidden"
                        >
                          Kuna kitu umekosea hapa
                        </p>
                      </ValidationSubscript>
                    </div>
                  </FormInput>
                  <FormInput
                    inputName="password"
                    inputType="password"
                    label="Password"
                    onChange={[Function]}
                    placeholder="Enter password"
                  >
                    <div
                      className="form-group"
                    >
                      Password
                      <input
                        className="form-control"
                        name="password"
                        onChange={[Function]}
                        placeholder="Enter password"
                        required={true}
                        type="password"
                      />
                      <ValidationSubscript>
                        <p
                          className="subscript subscript-warning hidden"
                        >
                          Kuna kitu umekosea hapa
                        </p>
                      </ValidationSubscript>
                    </div>
                  </FormInput>
                  <FormInput
                    inputName="repeatPassword"
                    inputType="password"
                    label="repeatPassword"
                    onChange={[Function]}
                    placeholder="Retype the password"
                  >
                    <div
                      className="form-group"
                    >
                      repeatPassword
                      <input
                        className="form-control"
                        name="repeatPassword"
                        onChange={[Function]}
                        placeholder="Retype the password"
                        required={true}
                        type="password"
                      />
                      <ValidationSubscript>
                        <p
                          className="subscript subscript-warning hidden"
                        >
                          Kuna kitu umekosea hapa
                        </p>
                      </ValidationSubscript>
                    </div>
                  </FormInput>
                  <button
                    className="btn btn-success btn-block"
                    onClick={[Function]}
                    type="submit"
                  >
                    Submit
                  </button>
                  <hr />
                  Already have an account?
                  <NavLink
                    activeClassName="active"
                    aria-current="page"
                    to="/"
                  >
                    <Route
                      path="\\\\/"
                    >
                      <Link
                        aria-current="page"
                        className="active"
                        replace={false}
                        style={Object {}}
                        to="/"
                      >
                        <a
                          aria-current="page"
                          className="active"
                          href="/"
                          onClick={[Function]}
                          style={Object {}}
                        >
                          Login
                        </a>
                      </Link>
                    </Route>
                  </NavLink>
                </form>
              </div>
            </div>
          </div>
        </RegistrationForm>
      </Connect(RegistrationForm)>
    </Router>
  </MemoryRouter>
</Provider>
`;
